### ISSUE TITLE
String interpolation creates false positive RS1032

### ISSUE THREAD

**Issue by zlepper** [dotnet/roslyn-analyzers#5717]
### Analyzer

**Diagnostic ID**: [RS1032](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1032-define-diagnostic-message-correctly): `Define diagnostic message correctly`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

Also happens with version 5.0.3 of the nuget package

### Describe the bug

The code
```c#

        public static readonly DiagnosticDescriptor MissingPropertyAnnotation =
            new("DZ00102", "Missing property annotation",
                $"A property of a {nameof(BaseDamResource)} must be annotated with either {nameof(DamResourceFieldAttribute)} or {nameof(PrivateDamResourceFieldAttribute)}",
                "ERROR", DiagnosticSeverity.Error, true);
```
Triggers RS1032, even though the text conforms. 

### Steps To Reproduce
See "Describe the bug"
### Expected behavior
This should not trigger the analyzer error

### Actual behavior
RS1032 is triggerd
